test/quantization/test_quant_primitives.py

```diff
@@ -102,7 +102,8 @@ def _groupwise_affine_quantize_tensor_from_qparams(
         .reshape_as(w)
     )
     if TORCH_VERSION_AT_LEAST_2_5:
-        w_int4x8 = (w_int4x8[::, ::2] << 4 | w_int4x8[::, 1::2]).to(torch.uint8)
+        if w.device.type != "cpu":
```

Comment:
maybe you can use https://github.com/pytorch/ao/blob/39f16f49f1c4a99e50cea55ba367322a617ce077/torchao/dtypes/utils.py#L60

Reply:
Done.